lib/client.js

```diff
@@ -232,6 +232,28 @@ var Client = function(options) {
   this._context = options.context || null;
   this.ready = false;
 
+  var WHITELISTED_ORIGINS = [
```

Comment:
I'm guessing it was a conscious choice to make the Regexp literals simple so as to be unambiguous?  (Could have put all of them into a single Regexp with optional groups, then just avoided the loop below)